src/main/java/edu/ucsb/cs156/frontiers/controllers/RosterStudentsController.java

```diff
@@ -267,4 +268,25 @@ public RosterStudent updateRosterStudent(
 
         return rosterStudentRepository.save(rosterStudent);
     }
+
+    @Operation(summary = "Delete a roster student")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("/delete")
+    @Transactional
+    public ResponseEntity<String> deleteRosterStudent(@Parameter(name = "id") @RequestParam Long id) throws EntityNotFoundException{
+        RosterStudent rosterStudent = rosterStudentRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RosterStudent.class, id));
+        Course course = rosterStudent.getCourse();
+        if(course != null){
+            List<RosterStudent> students = course.getRosterStudents();
+            if(students != null){
+                students.remove(rosterStudent);
+                course.getRosterStudents().clear();
+                course.getRosterStudents().addAll(students);
+                courseRepository.save(course);
+            }
+        }
```

Comment:
Is this really necessary?   I think just deleting the student from the rosterStudentRepository will have the same effect and be much simpler!

Does this not work?

```suggestion
        rosterStudentRepository.deleteById(id);
```

Reply:
That should greatly simply the tests too.

Reply:
The reason why I had these lines of codes is that I wanted to make sure if I delete a student from the student then this student should be also removed from the course which he/she is originally enrolled in. If I delete these codes, even though the student is removed from the repository, he/she will still remain in the course.

Reply:
Yes I'm sure. I made the suggested changes and tested DELETE on localhost Swagger, and that won't delete a student from the course.

Reply:
To be more precise, by "that won't delete a student from the course", I meant "it won't delete a student from the Student Roster of a course (accessed by id)".

Reply:
@linzezhou As it turns out, you are exactly right; I apologize for doubting you.

I've gone down a deep dive on this.   

There is, though, perhaps a cleaner way to disassociate the roster student from the course.  Can we try this?

```
Course course = rosterStudent.getCourse();
course.getRosterStudents().remove(rosterStudent);
rosterStudentRepository.delete(rosterStudent);
courseRepository.save(course); // optional, ensures persistence context is aware
```